lib/shelley/bench/restore-bench.hs

```diff
@@ -596,6 +614,52 @@ benchmarksSeq _ w wid _wname benchname restoreTime = do
         , walletOverview
         }
 
+data BenchBaselineResults = BenchBaselineResults
+    { benchName :: Text
+    , restoreTime :: Time
+    } deriving (Show, Generic)
+
+instance Buildable BenchBaselineResults where
+    build = genericF
+
+instance ToJSON BenchBaselineResults where
+    toJSON = genericToJSON Aeson.defaultOptions
+
+{- HLINT ignore bench_baseline_restoration "Use camelCase" -}
+bench_baseline_restoration
+    :: forall (n :: NetworkDiscriminant) .
+        (
+         NetworkDiscriminantVal n
+        , HasNetworkId n
+        )
+    => Proxy n
+    -> Tracer IO (BenchmarkLog n)
+    -> Trace IO Text -- ^ For wallet tracing
+    -> CardanoNodeConn  -- ^ Socket path
+    -> NetworkParameters
+    -> NodeToClientVersionData
+    -> Text -- ^ Benchmark name (used for naming resulting files)
+    -> Bool -- ^ if to write to disk
+    -> Percentage -- ^ Target sync progress
+    -> IO SomeBenchmarkResults
+bench_baseline_restoration proxy tr wlTr socket np vData benchname traceToDisk  targetSync = do
+    putStrLn $ "*** " ++ T.unpack benchname
+    let networkId = networkIdVal proxy
+    let gp = genesisParameters np
+    withWalletLayerTracer benchname traceToDisk $ \progressTrace -> do
+        withNetworkLayer (trMessageText wlTr) networkId np socket vData sTol $ \nw -> do
+            wait <- newEmptyTMVarIO
+            synchronizer <- async $ chainSync nw nullTracer $ ChainFollower
+                do pure []
+                do \blocks _ntip -> do
+                    let headers =  header . fromCardanoBlock gp <$>  blocks
```

Comment:
As discussed, we'd like to use ledger types directly instead of `fromCardanoBlock` here, so that we get a baseline that does not depends on our types (which may or may not be inefficient).

Reply:
Working around compatibility ended up in rewriting it. I'm not sure we are going anywhere here

Reply:
Thanks! I was worried that the conversions to our types might introduce inefficiencies — `length txs'` and `length .  map fromShelleyTx $ toList txs'` are different, lazy evaluation is different, etc. . But there is no need to go any further, indeed.